File: options/GlobalOptions.swift
//
//  PeripheralOptions.swift
//  react-native-bluetooth-le
//
//  Created by Andrew Tkachuk on 08.01.2024.
//

class GlobalOptions: NSObject {
    var timeoutDuration: Int = Constants.DEFAULT_TIMEOUT
    var autoDecode = false
    
    init(options: NSDictionary?) {
        if (options == nil) {
            return
        }
        
        timeoutDuration = options?[keys.timeoutDuration] as? Int ?? Constants.DEFAULT_TIMEOUT
        autoDecode = options?[keys.autoDecodeBytes] as? Bool ?? false
    }
    
    struct keys {
        static let timeoutDuration = "timeoutDuration"
        static let autoDecodeBytes = "autoDecodeBytes"
    }
}

File: android/src/main/java/com/bluetoothle/GlobalOptions.kt
package com.bluetoothle

import com.bluetoothle.GlobalOptions.Companion.Keys.autoDecodeBytes
import com.bluetoothle.constants.Constants
import com.facebook.react.bridge.ReadableMap

class GlobalOptions(options: ReadableMap?) {
  var autoDecode = false
  var timeoutDuration = Constants.DEFAULT_TIMEOUT

  init {
    if (options != null) {
      autoDecode = getAutoDecode(options)
      timeoutDuration = getTimeout(options)
    }
  }

  private fun getTimeout(options: ReadableMap): Int {
    if (options.hasKey(Keys.timeoutDuration)) {
      return options.getInt(Keys.timeoutDuration)
    }

    return Constants.DEFAULT_TIMEOUT
  }

  private fun getAutoDecode(options: ReadableMap): Boolean {
    if (options.hasKey(autoDecodeBytes)) {
      return options.getBoolean(autoDecodeBytes)
    }

    return false
  }

  companion object {
    object Keys {
      const val autoDecodeBytes = "autoDecodeBytes"
      const val timeoutDuration = "timeoutDuration"
    }
  }
}
